base/src/main/java/io/spine/query/RecordQuery.java

```diff
@@ -84,4 +86,47 @@
     public final RecordQueryBuilder<I, R> toBuilder() {
         return builder;
     }
+
+    /**
+     * Appends a series of record column predicates to this query treating them in conjunction
+     * with those predicates which are already set for querying.
+     *
+     * <p>This method only processes the column predicates. Additional identifier conditions,
+     * field masks, sorting, or limit are ignored.
+     */
+    public final RecordQuery<I, R> and(RecordPredicates<I, R> builder) {
+        RecordQuery<I, R> result = joinToRootPredicate(builder, AND);
+        return result;
+    }
+
+    /**
+     * Appends a series of records column predicates to this query treating them in disjunction
+     * with those predicates which are already set for querying.
+     *
+     * <p>This method only processes the column predicates. Additional identifier conditions,
+     * field masks, sorting, or limit are ignored.
+     */
+    public final RecordQuery<I, R> either(RecordPredicates<I, R> predicates) {
```

Comment:
Let's use `and` and `or`. Or maybe `all` and `either`, but not mix the two options.

Reply:
Semantically, this `Query.either(..)` is a shortcut to `Query.toBuilder().either(..).build()`. And while these `either()`s are methods of two different types — the former belongs to `Query`, the latter belongs to `QueryBulder` — it would be nice to use the same word.

Therefore, as discussed vocally, we will keep `either` here.